lib/webbluetooth/bindings.js

```diff
@@ -0,0 +1,298 @@
+var events = require('events');
+var util = require('util');
+
+var debug = require('debug')('bindings');
+
+var ble = navigator.bluetooth;
+
+//stub data until web APIs are implemented.
+var tempResponses = require('./tempResponses');
+
+
+function addDashes(uuid){
+  if(uuid && uuid.length === 32){
+    uuid = uuid.substring(0,8) + '-' + uuid.substring(8,12) + '-' +  uuid.substring(12,16) + '-' +  uuid.substring(16,20) + '-' +  uuid.substring(20);
+  }
+  return uuid.toLowerCase();
+}
+
+function stripDashes(uuid){
+  if(uuid){
+    uuid = uuid.split('-').join('');
+  }
+  return uuid;
+}
+
+
+var NobleBindings = function() {
+
+  this._startScanCommand = null;
+  this._peripherals = {};
+
+  var self = this;
+
+  process.nextTick(function(){
+    console.log('first state change');
+    self.emit('stateChange', 'poweredOn');
+  });
+  
+};
+
+util.inherits(NobleBindings, events.EventEmitter);
+
+NobleBindings.prototype.init = function() {
+  var self = this;
+  // no-op
+  setTimeout(function(){
+    console.log('initing');
+    self.emit('stateChange', 'poweredOn');
+  }, 2000);
+};
+ 
+
+NobleBindings.prototype._onOpen = function() {
+  console.log('on -> open');
+};
+
+NobleBindings.prototype._onClose = function() {
+  console.log('on -> close');
+
+  this.emit('stateChange', 'poweredOff');
+};
+
+NobleBindings.prototype.startScanning = function(serviceUuids, allowDuplicates) {
```

Comment:
@monteslu Did you figure this one out?
See for instance https://googlechrome.github.io/samples/web-bluetooth/read-characteristic-value-changed.html where `battery_service` is defined in `optionalServices`.